File: library/src/main/java/com/hansoolabs/and/view/SimpleRecyclerViewCalculator.java
package com.hansoolabs.and.view;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

/**
 * Created by brownsoo on 2017. 8. 23..
 */

public class SimpleRecyclerViewCalculator implements RecyclerViewCalculator {

    public int calculatePositionFromScrollProgress(@NonNull RecyclerView recyclerView,
                                                   float progress) {
        RecyclerView.Adapter<?> adapter = recyclerView.getAdapter();
        return Math.round((adapter.getItemCount() - 1) * progress);
    }

    public float calculateScrollProgress(@NonNull RecyclerView recyclerView) {
        int scrollTop = recyclerView.computeVerticalScrollOffset() +
                recyclerView.computeHorizontalScrollExtent();
        int maxScrollTop = recyclerView.computeVerticalScrollRange();
        return scrollTop / (float) maxScrollTop;
    }
}

File: library/src/main/kotlin/com/hansoolabs/and/view/SimpleRecyclerViewCalculator.kt
package com.hansoolabs.and.view

import androidx.recyclerview.widget.RecyclerView

/**
 * Created by brownsoo on 2017. 8. 23..
 */

class SimpleRecyclerViewCalculator : RecyclerViewCalculator {

    override fun calculatePositionFromScrollProgress(recyclerView: RecyclerView,
                                                     progress: Float): Int {
        val adapter = recyclerView.adapter
        return Math.round((adapter.itemCount - 1) * progress)
    }

    override fun calculateScrollProgress(recyclerView: RecyclerView): Float {
        val scrollTop = recyclerView.computeVerticalScrollOffset() + recyclerView.computeHorizontalScrollExtent()
        val maxScrollTop = recyclerView.computeVerticalScrollRange()
        return scrollTop / maxScrollTop.toFloat()
    }
}

